ui/src/app/utils/isKeyOfObject.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Narrow a string to whether it is a valid key for an object or not.
+ */
+export const isKeyOfObject = <P extends string, T extends Record<P, T[P]>>(
+  prop: P,
+  object: T
+): object is T & Record<P, string> => {
```

Comment:
This is minor, but it seems a bit weird that this util is called `isKeyOfObject` but the generics use different names/letters. Also do we need the type predicate here? Wouldn't `boolean` as the return type suffice? I'm just wondering what the purpose of it is.

Reply:
Updated the variables and generic names. The reason this uses a type predicate is so that it can narrow a union of objects. I've added a comment explaining that.

Reply:
Right, so it is necessary. I guess it just reads weird because the function name makes it seem like we're just checking that `key` exists in `object`, but the result of the function ends up narrowing the `object` type. `objectHasKey` might be more indicative? Anyway semantics is boring and this is actually fine